src/ConnectedMode/Migration/Wizard/MigrationWizardWindow.xaml.cs

```diff
@@ -59,6 +80,18 @@ private void MigrationFinished()
             dialogResult = true;
         }
 
-        private void OnClosing(object sender, CancelEventArgs e) => this.DialogResult = dialogResult;
+        private void OnClosing(object sender, CancelEventArgs e)
+        {
+            migrationInProgress = false;
+            this.DialogResult = dialogResult;
+        }
+
+        void IProgress<MigrationProgress>.Report(MigrationProgress value)
+        {
+            ListBoxItem item = new ListBoxItem();
```

Comment:
How does this work from a threading point of view? Presumably this update needs to be on the UI thread.
Does the whole process take place on the UI thread i.e. does the migration start on the UI thread?
(not necessarily a problem now, but it might be when we start doing "real work" during the migration).

Reply:
hmm yea good point, probably we have to add some thread handling. Should I do it in this pr already?

Reply:
No, just a new note on the board.